barretenberg/cpp/src/barretenberg/vm/avm/recursion/goblin_avm_recursive_verifier.hpp

```diff
@@ -0,0 +1,201 @@
+#pragma once
+
+#include "barretenberg/circuit_checker/circuit_checker.hpp"
+#include "barretenberg/goblin/goblin.hpp"
+#include "barretenberg/numeric/random/engine.hpp"
+#include "barretenberg/stdlib/goblin_verifier/goblin_recursive_verifier.hpp"
+#include "barretenberg/stdlib/hash/poseidon2/poseidon2.hpp"
+#include "barretenberg/stdlib/translator_vm_verifier/translator_recursive_verifier.hpp"
+#include "barretenberg/stdlib_circuit_builders/mega_flavor.hpp"
+#include "barretenberg/stdlib_circuit_builders/ultra_rollup_flavor.hpp"
+#include "barretenberg/stdlib_circuit_builders/ultra_rollup_recursive_flavor.hpp"
+#include "barretenberg/ultra_honk/decider_proving_key.hpp"
+#include "barretenberg/ultra_honk/ultra_prover.hpp"
+#include "barretenberg/ultra_honk/ultra_verifier.hpp"
+#include "barretenberg/vm/avm/generated/circuit_builder.hpp"
+#include "barretenberg/vm/avm/generated/composer.hpp"
+#include "barretenberg/vm/avm/recursion/recursive_flavor.hpp"
+#include "barretenberg/vm/avm/recursion/recursive_verifier.hpp"
+#include "barretenberg/vm/avm/trace/common.hpp"
+#include "barretenberg/vm/avm/trace/helper.hpp"
+#include "barretenberg/vm/avm/trace/public_inputs.hpp"
+#include "barretenberg/vm/avm/trace/trace.hpp"
+namespace bb::avm {
+
+/**
```

Comment:
This is the main contribution of this PR. I've tried to describe the algorithm in sufficient detail here.